internal/metrics/db.go

```diff
@@ -76,33 +75,85 @@ func FlushMetrics(ctx context.Context) error {
 	return nil
 }
 
-// / Spens up to 15 seconds sending all metrics collected so far to flyctl-metrics post endpoint
-func SendMetrics(ctx context.Context, json string) error {
-	authToken, err := GetMetricsToken(ctx)
+func SendMetrics(ctx context.Context, jsonData string) error {
+	cfg := config.FromContext(ctx)
+	metricsToken, err := GetMetricsToken(ctx)
 	if err != nil {
-		return err
+		fmt.Fprintf(os.Stderr, "Warning: Metrics token unavailable: %v\n", err)
+		return nil
 	}
 
-	cfg := config.FromContext(ctx)
-	request, err := http.NewRequest("POST", cfg.MetricsBaseURL+"/metrics_post", bytes.NewBuffer([]byte(json)))
+	baseURL := cfg.MetricsBaseURL
+	endpoint := baseURL + "/metrics_post"
+	userAgent := fmt.Sprintf("flyctl/%s", buildinfo.Info().Version)
+
+	errChan := make(chan error, 1)
+
+	go sendMetricsRequest(endpoint, metricsToken, userAgent, []byte(jsonData), errChan)
+
+	err = waitForCompletion(errChan)
 	if err != nil {
-		return err
+		fmt.Fprintf(os.Stderr, "Warning: Metrics send issue: %v\n", err)
+	}
+
+	return nil
+}
+
+func sendMetricsRequest(endpoint, token, userAgent string, data []byte, errChan chan<- error) {
+	request, err := http.NewRequest("POST", endpoint, bytes.NewBuffer(data))
+	if err != nil {
+		errChan <- fmt.Errorf("failed to create request: %w", err)
+		return
 	}
 
-	request.Header.Set("Authorization", authToken)
-	request.Header.Set("User-Agent", fmt.Sprintf("flyctl/%s", buildinfo.Info().Version))
+	request.Header.Set("Authorization", "Bearer "+token)
+	request.Header.Set("User-Agent", userAgent)
 
-	retryTransport := rehttp.NewTransport(http.DefaultTransport, rehttp.RetryAll(rehttp.RetryMaxRetries(3), rehttp.RetryTimeoutErr()), rehttp.ConstDelay(0))
+	client := createHTTPClient()
 
-	client := http.Client{
+	resp, err := client.Do(request)
+	if err != nil {
+		errChan <- fmt.Errorf("failed to send metrics: %w", err)
+		return
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		body, _ := io.ReadAll(resp.Body)
+		errChan <- fmt.Errorf("metrics send failed with status %d: %s", resp.StatusCode, string(body))
+		return
+	}
+
+	_, err = io.Copy(io.Discard, resp.Body)
+	if err != nil {
+		errChan <- fmt.Errorf("failed to read response body: %w", err)
+		return
+	}
```

Comment:
You don't need to drain the response body. You can just rely on the deferred `resp.Body.Close()` above.